sdcm/cluster.py

```diff
@@ -300,6 +300,51 @@ def __init__(self, name, parent_cluster, ssh_login_info=None, base_logdir=None,
         self._kernel_version = None
         self._uuid = None
         self.scylla_network_configuration = None
+        self._datacenter_name = None
+        self._node_rack = None
+
+    def _is_node_ready_run_scylla_commands(self) -> bool:
+        """
+        When node is just created and started to configure, during first node initializing, it is impossible to connect to the node yet and
+        `remoter` object is None. So we cannot connect to the node and run there needed commands.
+
+        When `remoter` is ready but cluster is not configured yet, `nodetool status` (and other Scylla commands) cannot be run.
+        Wait when it became available.
+        """
+        # During first cluster remoter is not available and may cause to failure
+        if not self.remoter:
+            self.log.warning("Remoter is not available")
+            return False
+
+        # During first cluster initializing the status info is not available and may cause to failure
+        if not self.db_up():
+            self.log.warning("Running Scylla commands are not available. Scylla cluster is not configured yet")
+            return False
+
+        return True
+
+    @property
```

Comment:
We should use cached_property, it would work better in this use case

Reply:
changed

Reply:
Actually it should be `property`, not `cached_property` because its value may be empty from beginning